### PR TITLE
support .gqlgen.yml

### PR DESCRIPTION
**Pull request by vvakame**
refs #161
I'm implemented types.yaml support 👍 

Now, We can processing 3 forms of `types.*` file.

old one.

```types.json
{
    "Element": "github.com/vektah/gqlgen/test.Element"
}
```

JSON form.

```types.json
{
    "Element": {
        "entityPath": "github.com/vektah/gqlgen/test.Element"
    }
}
```

YAML form.

```types.yaml
- typeName:   Element
  entityPath: github.com/vektah/gqlgen/test.Element
```

If json is used it may be better to display an error to use yaml.

### CODE REVIEW

**Comment by vvakame** [comment]
👀  https://github.com/vvakame/gqlgen/commit/26f865b32a71ba74c7e152cb1dde4d1311b969c5

```types.yaml
- typeName:   User
  entityPath: remote_api.User
  fields:
  - fieldName: likes
    resolver:  true
```

File: main.go [hunk 1]
```diff
@@ -54,20 +61,71 @@ func main() {
...
-			fmt.Fprintln(os.Stderr, "unable to open typemap: "+err.Error())
-			return nil
-		}
+func loadTypeMap() codegen.TypeMap {
+	if *typemap == "" {
+		return nil
+	}
+
+	if strings.HasSuffix(*typemap, ".json") {
+		return loadTypeMapJSON()
```
Comment (vektah) [inline main.go]: Yaml is a superset of json, so you should be able to remove the json codepath here entirely.

This also means you can use yaml comments in json files and leave on trailing newlines I think.

File: codegen/codegen.go [hunk 2]
```diff
@@ -34,6 +34,44 @@ type Config struct {
 	modelDir         string
 }
 
+type TypeMap []TypeMapEntry
```
Comment (vektah) [inline codegen/codegen.go]: This probably needs a better name now. Maybe it's just `Config`?

**Comment by vvakame** [comment]
@vektah thank for advice!

I added support about `.gqlgen.yml`.

```
schema: schema.graphql
output: generated.go
modeloutput: models/generated.go

models:
  Element:
    model: github.com/vektah/gqlgen/test.Element
  Viewer:
    model: github.com/vektah/gqlgen/test.Viewer
  User:
    model: remote_api.User
```

and, user use `-typemap` commandline option.
They will get the warning message likes below.

```
$ go run ../main.go -out generated.go -typemap types.json -models models/generated.go
you should use .gqlgen.yml with below content.

schema: schema.graphql
models:
  Element:
    model: github.com/vektah/gqlgen/test.Element
  User:
    model: remote_api.User
  Viewer:
    model: github.com/vektah/gqlgen/test.Viewer
output: generated.go
modeloutput: models/generated.go
```

**Pull request merged**